### DISCUSSION TITLE
Unable to render a new page for registration using mobile number instead of default registration flow

### DISCUSSION THREAD

**Discussion by Gayathri2102** [keycloak/keycloak#13341] [Q&A]
Hi Team,

I'm currently trying to modify the keycloak registration flow where I want the user to register their mobile number first and on authenticated via sms OTP, I want the user to input the email and password fields as second part of registration. I tried implementing the FormAction and FormActionFactory but it doesn't render my new template just before the default registration page. What is the best way to go about this flow?

**Comment by mrpatrickpilch** [comment]
Without having more details, you may be running into this bug: https://github.com/keycloak/keycloak-ui/issues/3727
